src/Pages/Top.elm

```diff
@@ -164,4 +169,41 @@ viewManifestList manifests =
 
 viewManifest : Manifest -> Element Msg
 viewManifest manifest =
-    row [] [ text manifest.shortName ]
+    row [ spacing 20
+        , padding 10
+        , width fill
+        ]
+        [ column []
+            [ image [ width (Element.px 24)]
+                { src = "../public/images/badge-outline-colored.svg"
+                , description = "App Icon goes here"
+                } ]
+        , column [ width fill ]
+            [ link
+                []
+                { url = Route.toString (Route.Preview__AppShortName_String {appShortName = manifest.shortName })
```

Comment:
We will want the link to go to the preview page on phones and the edit page on tablet and desktop devices. To determine the device, we need `model.device` to pattern match like we do in the `view` function. Problem is, we aren't currently passing it down to `viewManifest`. 

Here's how we can get it there:

- Update the type signature of `viewManifestList` to accept a `Device` and pass `model.device` to it in `view`
- Update the type signature of `viewManifest` to accept a `Device` and pass the device to it in `viewManifestList`. This one is a little trickier because we are mapping `viewManifest` over `manifests`. We can do it with partial application, which would look something like `List.map (viewManifest device) manifests`

That gets the device down to `viewManifest` and we can pattern match and link to the preview or edit page.

If you would like to try this out, go ahead and give it a shot. I could also make the change and you can review it if you'd prefer.

Reply:
I took a stab at it... let me know if that's what you had in mind :)